python/tests/test_array.py

```diff
@@ -1039,7 +1039,7 @@ def test_memoryless_copy(self):
 
         # Check that we get read-only array that does not own the underlying data
         self.assertFalse(a_np.flags.owndata)
-        self.assertFalse(a_np.flags.writeable)
+        self.assertTrue(a_np.flags.writeable)
```

Comment:
numpy seems to prefer the buffer protocol overs `__array__` internally. Since the buffer is writable, an array view is writable as well.